src/Tools/dotnet-monitor/Startup.cs

```diff
@@ -30,15 +28,14 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddMvc((MvcOptions options) =>
-            {
-                options.Filters.Add(new ProducesAttribute("application/json"));
-
-                // HACK We need to disable EndpointRouting in order to run properly in 3.1
-                System.Reflection.PropertyInfo prop = options.GetType().GetProperty("EnableEndpointRouting");
-                prop?.SetValue(options, false);
-            }).SetCompatibilityVersion(CompatibilityVersion.Latest);
+            services.AddMvc(options =>
+                {
+                    options.Filters.Add(new ProducesAttribute("application/json"));
 
+                    options.EnableEndpointRouting = false;
+                })
+                .SetCompatibilityVersion(CompatibilityVersion.Latest)
+                .AddApplicationPart(typeof(DiagController).Assembly);
```

Comment:
For the responsecompression, since we are now in 3.1, we can use brotli compression instead of gzip, which is better.

Reply:
Changed